modules/web/js/ballerina/diagram/views/default/components/resource-definition.jsx

```diff
@@ -44,20 +89,84 @@ class ResourceDefinition extends React.Component {
             || nodeFactory.isWorkerDeclaration(nodeBeingDragged);
     }
 
+    /**
+     * Creates the action menu.
+     * @memberof ResourceDefinition
+     */
+    createActionMenu() {
+        const model = this.props.model;
+        const designer = getDesigner(['default']);
+        const canvasOverlay = getCanvasOverlay();
+        // Recreating content
+        this.actionMenuWrapper = document.createElement('div');
```

Comment:
why are divs manually created instead of React creating them?

Reply:
canvasOverlay is an html node. There can be multiple actionMenuWrappers added to canvasOverlay added from multiple components(jsx) as each action menu can contain different menu items. If using react, only one actionMenuWrapper can be added to canvasOverlay using ReactDom. Cannot add multiple actionMenuWrappers using ReactDom to a canvasOverlay.